### PR TITLE
Navigator edits

### PR DESCRIPTION
**Pull request by PhillMyCup**

### CODE REVIEW

File: tests/auto/micromouse/navigator_test.h.autosave [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#ifndef NAVIGATOR_TEST_H
```
Comment (tbhova) [inline tests/auto/micromouse/navigator_test.h.autosave]: Delete this entire file.
Kevin probably has the autosave version still that needs saved

Comment (MrFobwatch) [reply tests/auto/micromouse/navigator_test.h.autosave]: Whoops. I'll do that shortly.

File: tests/auto/micromouse/navigator_test.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#include <vector>
+
+using namespace testing;
+using namespace std;
+
+class NavigatorTest : public ::testing::Test {
+    
+};
+
```
Comment (tbhova) [inline tests/auto/micromouse/navigator_test.h]: Add at least one test, something like:
```
TEST_F(NavigatorTest, testMapping) {

}
```

File: src/src/navigator.h [hunk 3]
```diff
@@ -11,18 +11,40 @@
  */
 
 class Navigator {
+private:
+    //Member Data
+    Driver driver;
+    GPS directions;
+    Maze maze;
+    Coordinate destination;
```
Comment (tbhova) [inline src/src/navigator.h]: I don't think this needs to be a member variable. Just make a new variable inside a function any time your need it.

File: src/src/navigator.h [hunk 4]
```diff
@@ -11,18 +11,40 @@
...
+    void updateMaze(Coordinate cell, std::vector<Cardinal8> walls); //Get wall data from Driver and adjust maze
```
Comment (tbhova) [inline src/src/navigator.h]: Change this to Doxygen Style documentation

File: src/src/navigator.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#include "navigator.h"
+using namespace std;
+Coordinate Navigator::findUnvisitedCell() {
+    for (int x = 0; x < sizeX; x++) {
```
Comment (tbhova) [inline src/src/navigator.cpp]: sizeX should be maze.getSizeX()

File: src/src/navigator.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
+#include "navigator.h"
+using namespace std;
+Coordinate Navigator::findUnvisitedCell() {
+    for (int x = 0; x < sizeX; x++) {
+        for (int y = 0; y < sizeY; y++){
```
Comment (tbhova) [inline src/src/navigator.cpp]: same thing for sizeY

File: src/src/navigator.cpp [hunk 7]
```diff
@@ -0,0 +1,51 @@
+#include "navigator.h"
+using namespace std;
+Coordinate Navigator::findUnvisitedCell() {
+    for (int x = 0; x < sizeX; x++) {
+        for (int y = 0; y < sizeY; y++){
+            if (!maze.hasMouseVisited(x,y)){
+                return Coordinate;
+                break;
```
Comment (tbhova) [inline src/src/navigator.cpp]: No break needed since we return

File: src/src/navigator.cpp [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+    }
+}
+
+
+
+bool Navigator::isDestination(int x, int y) {
+    return isCenter(x) && isCenter(y);
```
Comment (tbhova) [inline src/src/navigator.cpp]: isCenter is not defined anywhere in this cpp file

File: src/src/navigator.h [hunk 9]
```diff
@@ -11,18 +11,40 @@
...
+
+    /**
+     * @brief findUnvisitedCell - determine the Coordinates of the neares UnvisitedCell
+     * @return Coordinates of the the nearest UnvisitedCell
+     */
+    Coordinate findUnvisitedCell();
+        //Search maze for a cell.visited = false
```
Comment (tbhova) [inline src/src/navigator.h]: These comments can either be deleted or added to the method's @brief

File: src/src/navigator.h [hunk 10]
```diff
@@ -11,18 +11,40 @@
...
+        //determine coordinates of cell
+
+    bool isDestination(Coordinate cell); //Returns whether a cell is the destination
+    bool isCenter(int i);
```
Comment (tbhova) [inline src/src/navigator.h]: This method needs defined in the cpp

File: src/src/navigator.h [hunk 11]
```diff
@@ -11,18 +11,40 @@
...
 public:
     Navigator();
+
+    //this will be called by main - findUnvisitedCell with maze,
+    //use GPS to find direction to that cell, inform driver of direction
+    //Loop this until all cells are visited
```
Comment (tbhova) [inline src/src/navigator.h]: Change to Doxygen style

**Review by tbhova — CHANGES_REQUESTED** [review]
Address these comments and submit a new pull request

File: src/src/navigator.cpp (hunk 6 above)
Comment (MrFobwatch) [reply src/src/navigator.cpp]: Resolved

File: src/src/navigator.cpp (hunk 5 above)
Comment (MrFobwatch) [reply src/src/navigator.cpp]: Resolved

File: src/src/navigator.h (hunk 10 above)
Comment (MrFobwatch) [reply src/src/navigator.h]: Resolved

File: src/src/navigator.cpp (hunk 7 above)
Comment (MrFobwatch) [reply src/src/navigator.cpp]: Resolved

File: tests/auto/micromouse/navigator_test.h (hunk 2 above)
Comment (MrFobwatch) [reply tests/auto/micromouse/navigator_test.h]: Resolved

File: src/src/navigator.cpp (hunk 8 above)
Comment (MrFobwatch) [reply src/src/navigator.cpp]: Resolved

File: src/src/navigator.h (hunk 9 above)
Comment (MrFobwatch) [reply src/src/navigator.h]: Resolved

File: src/src/navigator.h (hunk 11 above)
Comment (MrFobwatch) [reply src/src/navigator.h]: Resolved

File: src/src/navigator.h (hunk 3 above)
Comment (MrFobwatch) [reply src/src/navigator.h]: Resolved

**Pull request closed without merging**